lib/plug/csrf_protection.ex

```diff
@@ -103,17 +103,34 @@ defmodule Plug.CSRFProtection do
 
   def init(opts), do: opts
 
-  def call(conn, _opts) do
+  def call(conn, opts) do
+    opts = verify_opts opts
     csrf_token = get_session(conn, "_csrf_token")
     Process.put(:plug_csrf_token, csrf_token)
-
     if not verified_request?(conn, csrf_token) do
-      raise InvalidCSRFTokenError
+      if opts[:with] == :exception do
```

Comment:
We could extract lines 111-115 here to a `handle_unverified_request(conn, opts)` function. Instead of using `opts[:with] == :exception`, we could do `if opts[:clear_session] do`. That way, users could do `plug Plug.CSRFProtection clear_session: true` instead of `plug Plug.CSRFProtection with: :nil_session`, which I think is clearer.

What do you think?